Lib/asyncio/unix_events.py

```diff
@@ -1381,8 +1381,8 @@ def _join_threads(self, timeout=None):
             thread.join(timeout)
 
         # Clear references to terminated threads
-        self.threads[:] = [thread for thread in list(self._threads.values())
-                           if thread.is_alive() and not thread.daemon]
+        self._threads[:] = [thread for thread in list(self._threads.values())
+                            if thread.is_alive() and not thread.daemon]
```

Comment:
As I was trying to let you see for yourself, `self._threads` is a dict, not a list. If you want to update it using a comprehension you can use something like this:
```
self._threads = {key: thread for key, thread in self._threads.items() if thread.daemon or thread.is_alive()}
```
Please test locally before merging.